sql/core/src/main/scala/org/apache/spark/sql/execution/python/EvalPythonUDTFExec.scala

```diff
@@ -99,7 +100,19 @@ trait EvalPythonUDTFExec extends UnaryExecNode {
         projection(inputRow)
       }
 
-      val outputRowIterator = evaluate(argMetas, projectedRowIter, schema, context)
+      memoryConsumer.foreach { consumer =>
+        val acquireMemoryMbActual: Long = consumer.acquireMemory()
+        udtf.acquireMemoryMbActual = Some(acquireMemoryMbActual)
+      }
+      val outputRowIterator = try {
+        evaluate(argMetas, projectedRowIter, schema, context)
+      } finally {
+        if (TaskContext.get() != null) {
```

Comment:
Why do we need this check?

Reply:
This was left-over from before I moved the check into the `memoryConsumer` object itself:

```
  lazy val memoryConsumer: Option[PythonUDTFMemoryConsumer] = {
    if (TaskContext.get() != null) {
      Some(PythonUDTFMemoryConsumer(udtf))
    } else {
      None
    }
  }
```

I removed this check here now.